### ISSUE TITLE
torch.quasirandom.SobolEngine(..., scramble=True) does not respect torch.manual_seed

### ISSUE THREAD

**Issue by mshvartsman** [pytorch/pytorch#36341]
## 🐛 Bug

`SobolEngine` does not respect `torch.manual_seed` when `seed = None` and `scramble=True` (otherwise it's fine). I think the easy fix would be to remove the `g.seed()` call at https://github.com/pytorch/pytorch/blob/master/torch/quasirandom.py#L64 but I'm not sure if there's something subtle I'm missing. 

## To Reproduce
```
import torch
from torch.quasirandom import SobolEngine

seed = 123454321
x1 = SobolEngine(dimension=1, scramble=True, seed=seed).draw(3)
x2 = SobolEngine(dimension=1, scramble=True, seed=seed).draw(3)

assert torch.all(x1 == x2) ## succeeds

# seeded via torch.manual_seed
torch.manual_seed(seed)
x1 = SobolEngine(dimension=1, scramble=True).draw(3)
torch.manual_seed(seed)
x2 = SobolEngine(dimension=1, scramble=True).draw(3)

assert torch.all(x1 == x2) ## fails
```
## Expected behavior

As per https://pytorch.org/docs/stable/notes/randomness.html, expected behavior is deterministic when setting seed except for some cases involving CUDA. 

## Environment

 - PyTorch Version (e.g., 1.0):1.4.0
 - OS (e.g., Linux): tested on OSX and CentOS
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): N/A
 - Python version: 3.7.6
 - CUDA/cuDNN version: N/A (cpu only)
 - GPU models and configuration: N/A

cc @ezyang @SsnL @gchanan @Balandat @vishwakftw.

**Comment by mshvartsman** [comment]
Correction, merely removing the seed call will make it always return the same values. I think the right thing may be to replace with `g = None` instead of `g.seed()`.

**Comment by Balandat** [comment]
> I think the right thing may be to replace with g = None instead of g.seed().

I agree, just tested this locally, works just fine. 

I support this change for better consistency. Since this is a backward-incompatible one, we'll need to make sure to announce it as such.

**Comment by Balandat** [comment]
I think we can close this now?
